rfcs/experimental/dynamic_selection/README.md

```diff
@@ -0,0 +1,516 @@
+
+<a id="submission_req_id"></a>
+### Submission
+
+The type `T` satisfies *Submission* for a given [Policy](#policy_req_id) `p` if given,
+
+- `s` an arbitrary identifier of type `T`
+
+| *Must* be well-formed | Description |
+| --------------------- | ----------- |
+| `s.get_policy()` | Returns an object that satisfies [Policy](#policy_req_id) and corresponds to the Policy that made the selection. |
+| `s.wait()` | Blocks until the submission has completed. |
+| `s.unwrap()` | Returns the underlying backend type value. This type may be void, may represent the backend’s synchronization type, or may represent a return value from the submission. |
+
+## Provided Concrete Policies 
+
+<a id="fixed_resource_id"></a>
+### `fixed_resource_policy`
+
+```cpp
+template<typename Backend=sycl_backend> 
+class fixed_resource_policy;
+```
+
+| Constructors and Initialization |
+| -----------------------------------------------------------|
+| `fixed_resource_policy(deferred_initialization_t); // (1)` |
+| `fixed_resource_policy(size_t offset=0); // (2)` |
+| `fixed_resource_policy(const std::vector<resource_t<Backend>>& resources, size_t offset=0); // (3)` |
+| `void initialize(size_t offset=0); // (4)` |
+| `void initialize(const std::vector<resource_t<Backend>>& resources, size_t offset=0); // (5)` |
+
+1. Defers initialization and requires a later call to `initialize`.
+2. Always selects index `offset` in the default set of resources.
+3. Uses the provided set of resources and always selects index `offset` from that set.
+4. Always selects index `offset` in the default set of resources.
+5. Uses the provided set of resources and always selects index `offset` from that set.
+
+`initialize` should only be called for policies that are constructed with `deferred_initialization_t`.
+
+#### select heuristic (expository)
+
+```cpp
+  template<typename ...Args>
+  selection_type fixed_resource_policy::select(Args&&...) {
+    if (initialized_) {
+      return selection_type{*this, resources_[fixed_offset_]};
+    } else {
+      throw std::logic_error(“select called before initialize”);
+    }
+  }
+```
+
+#### execution info reporting requirements
+
+none
+
+#### Exceptions
+
+Constructor or initialize may throw `std::bad_alloc` or `std::logic_error`.
+
+<a id="round_robin_id"></a>
+### `round_robin_policy`
+
+```cpp
+template<typename Backend=sycl_backend> 
+class round_robin_policy;
+```
+
+| Constructors and Initialization |
+| -----------------------------------------------------------|
+| `round_robin_policy(deferred_initialization_t); // (1)` |
+| `round_robin_policy(size_t offset=0); // (2)` |
+| `round_robin_policy(const std::vector<resource_t<Backend>>& resources); // (3)` |
+| `void initialize(size_t offset=0); // (4)` |
+| `void initialize(const std::vector<resource_t<Backend>>& resources); // (5)` |
```

Comment:
It seems that a defaulted `offset` parameter is not is not actually implemented, but rather a default ctor, and `initialize()` with no args.

Reply:
ok.